### PR TITLE
MSn tree visualizer

### PR DESCRIPTION
**Pull request by robinschmid**
This PR is mixed with the isotope finder PR #441 which should be merged first before reviewing this.

All spectra of level MS2, MS3, MS4 from top to bottom in one chart each.
image

image
 
The different energies are easy to compare in the charts.
image

### CODE REVIEW

File: src/main/java/io/github/mzmine/util/scans/ScanUtils.java [hunk 1]
```diff
@@ -72,6 +78,11 @@
  */
 public class ScanUtils {
 
+  /**
+   * tolerance to compute and combine precursor m/z
+   */
+  public static final int DEFAULT_PRECURSOR_MZ_TOLERANCE = 100;
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/util/scans/ScanUtils.java:84]: where and how is this used, it seems oddly high?

File: src/main/java/io/github/mzmine/util/scans/ScanUtils.java [hunk 2]
```diff
@@ -1006,6 +1018,62 @@ public static int findTopDataPoint(@NotNull double intensityValues[]) {
...
+        }
+      }
+      // sort
+      Collections.sort(result);
+      result.forEach(PrecursorIonTree::sort);
+    }
+    return result;
+  }
+
+  private static long getMzKey(double precursorMZ) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/util/scans/ScanUtils.java:1073]: ah i get it. so this is the best solution? what about tzhe range maps we usually use?

if it's used this way, it's not really a tolerance as we usually use it. if it's only used here, maybe move it into this method or give it a more descriptive name.

Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/util/scans/ScanUtils.java:1073]: maybe like PRECURSOR_ACCURACY_IN_DECIMAL_DIGITS = 3 and use 
`Math.round(Math.pow(10, PRECURSOR_ACCURACY_IN_DECIMAL_DIGITS - 1) * precursorMz)`

**Comment by SteffenHeu** [comment]
also i'm missing a load/save test for this class.

**Comment by robinschmid** [comment]
@SteffenHeu added test for MSnInfo - should be ready now

**Pull request merged**